src/java/org/apache/cassandra/config/Converters.java

```diff
@@ -132,7 +145,7 @@ public Object convert(Object value)
      */
     public Object deconvert(Object value)
     {
-        if (value == null) return 0;
+        if (value == null) return null;
```

Comment:
this was dead code before, but returning 0 was never safe as you don't know if the inputType was int or not